Add ZHA best practices for avoiding pairing difficulties (#17466)
home-assistant/home-assistant.github.io: source/_integrations/zha.markdown

1. Reset your Zigbee devices according to the device instructions provided by the manufacturer (e.g., turn on/off lights up to 10 times, switches usually have a reset button/pin). It might take a few seconds for the devices to appear. You can click on **Show logs** for more verbose output.
1. Once the device is found, it will appear on that page and will be automatically added to your devices. You can optionally change its name and add it to an area (you can change this later). You can search again to add another device, or you can go back to the list of added devices.

> ### Best practices for avoiding pairing difficulties

Reviewer: Is it correct that this should be under "Adding devices"? Is under troubleshooting not a better place for this?

Author: I think that this should be under the "Adding devices" as these are things you always need to consider when adding Zigbee devices, but feel free to move it if you think that it fits better there. You can always move it later once this is merged.

Reviewer: I'll leave that decision to the codeowners

Reviewer 2: I think it makes more sense to have it under "adding devices" section.
Be, should the numbering be updated? Both are  1?

Author: Those numberings are not for the new section in this PR but for the old section above it. Regardless, the numberings in that othering section look correct when looking at the whole file.

1. Reset your Zigbee devices according to the device instructions provided by the manufacturer (e.g., turn on/off lights up to 10 times, switches usually have a reset button/pin). It might take a few seconds for the devices to appear. You can click on **Show logs** for more verbose output.
1. Once the device is found, it will appear on that page and will be automatically added to your devices. You can optionally change its name and add it to an area (you can change this later). You can search again to add another device, or you can go back to the list of added devices.

> #### Best practices for avoiding pairing difficulties

Reviewer: ```suggestion
### Best practices for avoiding pairing difficulties
```

titles should be the same or +/- as the previous title

Is it correct that this should be under "Adding devices"?
Is not under troubleshooting a better place?

> -	If possible try to pair your Zigbee devices in their intended final location, (and not pair it next to the Zigbee coordinator and then need to move it after). 

Reviewer: ```suggestion
- If possible try to pair your Zigbee devices in their intended final location, (and not pair it next to the Zigbee coordinator and then need to move it after). 
```

>   -	Pairing a Zigbee device next to the Zigbee coordinator and then moving it later can result in a dropped/lost connections or other issues. 

Reviewer: ```suggestion
  - Pairing a Zigbee device next to the Zigbee coordinator and then moving it later can result in dropped/lost connections or other issues. 
```

> - Zigbee devices that in the past have been connected to another Zigbee coordinator (hub/bridge/gateway/controller) can usually only be added/paired in ZHA after they have been restored to their factory reset settings. 

Reviewer: ```suggestion
```

https://deploy-preview-17466--home-assistant-docs.netlify.app/integrations/zha/#add-philips-hue-bulbs-that-have-previously-been-added-to-another-bridge

This is already documented on the same page.

> -	Zigbee devices that have in the past been connected used to a other Zigbee coordinator (hub/bridge/gateway/controller) can usually only be added/paired in ZHA after they have been restored to their factory reset settings. 

Reviewer: ```suggestion
- Zigbee devices that in the past have been connected to another Zigbee coordinator (hub/bridge/gateway/controller) can usually only be added/paired in ZHA after they have been restored to their factory reset settings. 
```

>     -	If the device you want to add is not brand new and as such never paired before then you always have to make sure to first manually reset the device to its factory default settings before you will be able add/pair it. 

Reviewer: ```suggestion
    - If the device you want to add is not brand new and as such never paired before then you always have to make sure to first manually reset the device to its factory default settings before you will be able to add/pair it. 
```

> -	Some battery-operated Zigbee devices are known to have problems with pairing if they have Low battery voltage. 

Reviewer: ```suggestion
- Some battery-operated Zigbee devices are known to have problems with pairing if they have Low battery voltage. 
```

>     -	Some people have reported replacing the battery on their newly received Xiaomi/Aqara devices solved pairing issues. 

Reviewer: ```suggestion
    - Some people have reported replacing the battery on their newly received Xiaomi/Aqara devices solved pairing issues. 
```

> -	Check that you have enough Zigbee router devices (also known as Zigbee signal repeaters or range extenders) and if you do not have any, invest and add some mains-powered devices that will work as Zigbee routers. 

Reviewer: ```suggestion
- Check that you have enough Zigbee router devices (also known as Zigbee signal repeaters or range extenders) and if you do not have any, invest and add some mains-powered devices that will work as Zigbee routers. 
```

>     -	Aim to start out with mains-powered devices before adding battery-operated devices as a "weak" Zigbee network mesh (e.g., the device is too far from the Zigbee coordinator or a Zigbee router) may prevent some devices from being paired. Zigbee router devices are also needed to increase the maximum of devices that can be connected to your Zigbee mesh network. 

Reviewer: ```suggestion
    - Aim to start out with mains-powered devices before adding battery-operated devices as a "weak" Zigbee network mesh (e.g., the device is too far from the Zigbee coordinator or a Zigbee router) may prevent some devices from being paired. Zigbee router devices are also needed to increase the maximum of devices that can be connected to your Zigbee mesh network. 
```

>     -	Note that some Zigbee devices are not fully compatible with all brands of Zigbee router devices. Xiaomi/Aqara devices are for example known not to work with Zigbee router devices from Centralite, General Electrics, Iris, Ledvance/OSRAM/ LIGHTIFY/Sylvania, Orvibo, PEQ, Securifi, and SmartThings/Samsung. Better results can usually be achieved by using mains-powered devices IKEA and Nue/3A Home or dedicated DIY routing devices based on Texas Instruments CC253x/CC26x2 and XBee Series 2/3 Zigbee radios. 

Reviewer: ```suggestion
    - Note that some Zigbee devices are not fully compatible with all brands of Zigbee router devices. Xiaomi/Aqara devices are for example known not to work with Zigbee router devices from Centralite, General Electrics, Iris, Ledvance/OSRAM/ LIGHTIFY/Sylvania, Orvibo, PEQ, Securifi, and SmartThings/Samsung. Better results can usually be achieved by using mains-powered devices IKEA and Nue/3A Home or dedicated DIY routing devices based on Texas Instruments CC253x/CC26x2 and XBee Series 2/3 Zigbee radios. 
```

> -	Be patient as the pairing of some Zigbee devices may require multiple attempts and you may sometimes need to try again and again. 

Reviewer: ```suggestion
- Be patient as the pairing of some Zigbee devices may require multiple attempts and you may sometimes need to try again and again. 
```

>     -	 Some devices, like example those from Xiaomi/Aqara, are known to not be 100% compliant with the standard Zigbee specifications and may therefor require many paring attempts over 10-20 minutes or longer. 

Reviewer: ```suggestion
    - Some devices, like example those from Xiaomi/Aqara, are known to not be 100% compliant with the standard Zigbee specifications and may therefore require many paring attempts over 10-20 minutes or longer. 
```
